component-sets/material/docs/src/content/docs/basic-widgets/text_field.mdx

```diff
@@ -0,0 +1,148 @@
+---
+<!-- Copyright © SixtyFPS GmbH <info@slint.dev> ; SPDX-License-Identifier: MIT -->
+title: TextField
+description: TextField API.
+---
+
+import SlintProperty from '/src/components/SlintProperty.astro';
+import CodeSnippetMD from '/src/components/CodeSnippetMD.astro';
+import Link from '/src/components/Link.astro';
+
+<CodeSnippetMD imagePath="/src/assets/generated/textfield-example.png" imageWidth="200" imageHeight="200" imageAlt='TextField example'>
+
+```slint playground
+import { TextField, Vertical } from "material.slint";
+export component Example inherits Window {
+    width: 200px;
+    height: 100px;
+
+    Vertical {
+        TextField {
+            label: "Name";
+            placeholder: "Enter your name";
+        }
+    }
+}
+```
+</CodeSnippetMD>
+
+A widget used to enter a single line of text with additional features like icons and supporting text.
+
+## Properties
+
+### label
+<SlintProperty propName="label" typeName="string">
+The label text displayed above the text input.
+</SlintProperty>
+
+### placeholder
+<SlintProperty propName="placeholder" typeName="string">
+A placeholder text being shown when there is no text in the edit field.
+</SlintProperty>
+
+### supporting_text
+<SlintProperty propName="supporting_text" typeName="string">
+Additional text displayed below the text input, often used for hints or error messages.
+</SlintProperty>
+
+### has_error
+<SlintProperty propName="has_error" typeName="bool">
+Indicates whether the text field should display an error state.
+</SlintProperty>
+
+### leading_icon
+<SlintProperty propName="leading_icon" typeName="image">
+An icon displayed at the beginning of the text input.
+</SlintProperty>
+
+### trailing_icon
+<SlintProperty propName="trailing_icon" typeName="image">
+An icon displayed at the end of the text input.
+</SlintProperty>
+
+### enabled
+<SlintProperty propName="enabled" typeName="bool" defaultValue="true">
+When false, the text field is disabled and cannot be interacted with.
+</SlintProperty>
+
+### has_focus
+<SlintProperty propName="has_focus" typeName="bool" propertyVisibility="out">
+Set to true when the text field currently has the focus.
+</SlintProperty>
+
+### text
+<SlintProperty propName="text" typeName="string" propertyVisibility="in-out">
+The text being edited.
+</SlintProperty>
+
+## Functions
+
+### set_selection_offsets(int, int)
+Selects the text between two UTF-8 offsets.
+
+### select_all()
+Selects all text in the text field.
+
+### clear_selection()
+Clears the current text selection.
+
+### cut()
+Copies the selected text to the clipboard and removes it from the text field.
+
+### copy()
+Copies the selected text to the clipboard.
+
+### paste()
+Pastes the text content of the clipboard at the cursor position.
+
+## Callbacks
+
+### leading_clicked()
```

Comment:
I'll bring this also up in the API review, but `leading` in the context of text has a very different meaning and is slightly confusing because it *could* imply a click into the leading of the text. If this is about the icon, then I think the name used in the spec makes most sense in the API here as well: "leading icon".